build-generator.sh

```diff
@@ -34,5 +34,5 @@ trap 'onError' ERR
 echo "Building generator"
 
 cd "${BASE_DIR}/generator"
-go generate ./...
+go generate ./..
```

Comment:
This looks strange to me -- `...` has special meaning in go land (wildcard for packages), whereas I think `./..` will just be interpreted as a path.

Reply:
@amisevsk thanks for the hint :)

I've followed the same workaround proposed here (also the reason of the error I was experiencing): https://github.com/golang/mock/issues/415#issuecomment-602684505 which is a similar case of upgrading from `1.13` to `> 1.13`.

My issue previously was that when I was trying to run `go generate ./...` I was receiving the following error:
```
cannot find package "." in:
        /home/tpetkos/github.com/thepetk/devfiles/devfile-api/generator/vendor/sigs.k8s.io/controller-tools/cmd/helpgen
crds/gen.go:34: running "go": exit status 1
cannot find package "." in:
        <my-path>/generator/vendor/sigs.k8s.io/controller-tools/cmd/helpgen
getters/gen.go:31: running "go": exit status 1
cannot find package "." in:
       <my-path>/generator/vendor/sigs.k8s.io/controller-tools/cmd/helpgen
interfaces/gen.go:33: running "go": exit status 1
cannot find package "." in:
        <my-path>/generator/vendor/sigs.k8s.io/controller-tools/cmd/helpgen
overrides/gen.go:41: running "go": exit status 1
cannot find package "." in:
        <my-path>/generator/vendor/sigs.k8s.io/controller-tools/cmd/helpgen
schemas/gen.go:46: running "go": exit status 1
cannot find package "." in:
        <my-path>/generator/vendor/sigs.k8s.io/controller-tools/cmd/helpgen
validate/gen.go:32: running "go": exit status 1
```

As suggested in the similar case, the `-mod=mod` flag is fixing that issue and we are able again to use `go generate ./...`

Reply:
Hmm. I'm concerned that using `-mod=mod` makes the vendor directory meaningless (if I understand correctly, this will direct go to _not_ use our vendor directory, defeating the purpose of having it).

Looking at the error, it seems to be coming from controller-tools. I found the following issue that looks to be the same problem -- https://github.com/kubernetes-sigs/controller-tools/issues/327 -- though it was closed due to inactivity, not due to an actual fix.

Reply:
@amisevsk getting back to you after I've tried various stuff while changing the codebase the least possible (having in mind that the scope of this issue is the upgrade of the go version).

I think the issue you've mentioned is very similar with our issue here. What I've noticed is that we cannot vendor the `cmd/helpgen` directory as it is under `package main` so upon `go mod vendor` we only vendor the `pkg` directory. As a result we cannot find the `helpgen` module we are trying to run the `go:generate` commands we have across the `generator` directory.

What I've tried is to do is to add the `-mod=mod` flag while using a fixed version of helpgen (same with the one that the API is using). It might not be the best approach, but I thought maybe we could create a separate issue to resolve this in a better way in the future (as the `generator` atm uses a very old version of go). WDYT?

Reply:
I suppose it's the best we can do, makes sense.